libraries/matrix/src/main/kotlin/io/element/android/libraries/matrix/session/PreferencesSessionStore.kt

```diff
@@ -16,91 +16,62 @@
 
 package io.element.android.libraries.matrix.session
 
-import android.content.Context
-import androidx.datastore.core.DataStore
-import androidx.datastore.preferences.core.Preferences
-import androidx.datastore.preferences.core.edit
-import androidx.datastore.preferences.core.stringPreferencesKey
-import androidx.datastore.preferences.preferencesDataStore
 import com.squareup.anvil.annotations.ContributesBinding
+import com.squareup.sqldelight.runtime.coroutines.asFlow
+import com.squareup.sqldelight.runtime.coroutines.mapToOneOrNull
 import io.element.android.libraries.di.AppScope
-import io.element.android.libraries.di.ApplicationContext
 import io.element.android.libraries.di.SingleIn
-import io.element.android.libraries.matrix.core.SessionId
+import io.element.android.libraries.matrix.Database
+import io.element.android.libraries.matrix.core.UserId
 import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.firstOrNull
 import kotlinx.coroutines.flow.map
-import kotlinx.serialization.Serializable
-import kotlinx.serialization.decodeFromString
-import kotlinx.serialization.encodeToString
-import kotlinx.serialization.json.Json
 import org.matrix.rustcomponents.sdk.Session
 import javax.inject.Inject
 
-private val Context.dataStore: DataStore<Preferences> by preferencesDataStore(name = "elementx_sessions")
-
-// TODO It contains the access token, so it has to be stored in a more secured storage.
-private val sessionKey = stringPreferencesKey("session")
-
 @SingleIn(AppScope::class)
 @ContributesBinding(AppScope::class)
 class PreferencesSessionStore @Inject constructor(
```

Comment:
We should probably have this implementation in a separate module? 
It's not really the responsability of the matrix-module to know that. 
But we can do that later when we'll expose a matrix-api module

In the meantime, it should probably be named `SqliteSessionStore`